rdt/hyper_transformer.py

```diff
@@ -155,18 +194,28 @@ def fit(self, data):
             data (pandas.DataFrame):
                 Data to fit the transformers to.
         """
-        if self.transformers is not None:
-            self._transformers = load_transformers(self.transformers)
-        else:
-            self._transformers = self._analyze(data)
+        self._input_columns = list(data.columns)
+        self._update_field_types(data)
+        fitted_fields = set()
+
+        # Loop through field_transformers that are first level
+        for field in self.field_transformers:
```

Comment:
I think that should keep track of which input arguments were used during fit and which were not. If the user mistakenly passed a field name that ended up never being used we should issue a warning or raise an error (possibly just a warning):

```python
In [15]: ht = HyperTransformer(field_transformers={
    ...:     'something': SomeTransformer,
    ...:     'something_else': AnotherTransformer,
    ...:     })

In [16]: ht.fit(data)
<ipython-input-11-37d0f57b9965>:5: RuntimeWarning: Field "something_else" was specified in the input arguments was not found in the data
  warnings.warn('Field "something_else" was specified in the input arguments was not found in the data', RuntimeWarning)
```